### PR TITLE
Add CLI switch for audio buffer size and number

### PR DESCRIPTION
**Pull request by pachuco**
Format is -ab:PAGESIZE:NUMPAGES
Default is values 512:32.

I typically pass -ab:4096:2 to get emu running nicely on my end.

### CODE REVIEW

**Comment by pachuco** [comment]
I could have put it all in -a: instead, like -a:DEVICE:PAGESIZE:NUMPAGES, but it felt bad to force passing an audio device just to pass buffer size.
Unless device 0 is the default?

**Comment by Falcosoft** [comment]
Device 0 is NOT the OS default device. Device 0 is always the 1st device so if you change the default device e.g. on Win Vista+ sound control panel it does not reflect the change. You have to pass explicit NULL to SDL_OpenAudioDevice() as first parameter to use OS default device (and so in order the emulator to 'follow' the device selected in the sound control panel).

**Comment by pachuco** [comment]
Hmmm, another way I could see this done is passing -a:x:4096:2, where x tells to use default value.
Or maybe -a:d1:b4096:n2. Oh god, that looks abominable...

**Comment by pachuco** [comment]
Oh, oh, better idea, ipv6 style!
-a::4096:2 ommit audio device
-a:3:4096:2 all options
-a:3::24 ommit buffer size

Eh, still kinda ugly.

File: src/mcu.cpp [hunk 1]
```diff
@@ -1180,9 +1191,25 @@ int main(int argc, char *argv[])
             {
                 port = atoi(argv[i] + 3);
             }
-            else if (!strncmp(argv[i], "-a:", 3))
+            else if (!strncmp(argv[i], "-d:", 3))
```
Comment (nukeykt) [inline src/mcu.cpp:1194]: any reason to change this? and what benefit of doing strchr below?

File: src/mcu.cpp [hunk 2]
```diff
@@ -1180,9 +1191,25 @@ int main(int argc, char *argv[])
...
+                    audioDeviceIndex = atoi(++pColon);
+                } while (0);
+            }
+            else if (!strncmp(argv[i], "-ab:", 4))
             {
-                audioDeviceIndex = atoi(argv[i] + 3);
+                char* pColon = argv[i];
+                
+                do {
+                    if (!(pColon = strchr(pColon, ':')) || pColon[1] == 0) break;
```
Comment (nukeykt) [inline src/mcu.cpp:1208]: first strchr is redundant because we know that it is exactly on argv[i]+3

**Review by nukeykt — CHANGES_REQUESTED** [review]

File: src/mcu.cpp (hunk 1 above)
Comment (pachuco) [reply src/mcu.cpp:1194]: I first tried to merge -a with -ab. It didn't go very cleanly, so I separated them back, acvidentally turning -a to -d(from "device"). Oops...

File: src/mcu.cpp (hunk 2 above)
Comment (pachuco) [reply src/mcu.cpp:1208]: The do + strchr + break thing prevents passing things like -a:or -a.
Very clever! But maybe too clever when debugging later?

Comment (pachuco) [reply src/mcu.cpp:1203]: Let me know how you feel about the remaining shenanigans.

File: src/mcu.cpp [hunk 3]
```diff
@@ -1184,6 +1195,17 @@ int main(int argc, char *argv[])
             {
                 audioDeviceIndex = atoi(argv[i] + 3);
             }
+            else if (!strncmp(argv[i], "-ab:", 4))
+            {
+                char* pColon = argv[i] + 3;
+                
+                do {
```
Comment (nukeykt) [inline src/mcu.cpp:1202]: any reason to use do while(0) construction?

Comment (pachuco) [reply src/mcu.cpp:1202]: the do {} while(0) construction is so I can break; out of trying to parse values.
Range checks? I suppose checking if atoi() returns 0 is a sane one(letters/symbols passed to atoi). 0 is dangerous situation, you are right.

Also, I prefer if user makes most of range checks himself.
If insane value is passed and application behaves poorly, maybe insane value should not be passed!

Comment (nukeykt) [reply src/mcu.cpp:1202]: normal `if`` `else` can be used here though, no?
I'd prefer default value used if bad values are passed instead of crash or hang

**Pull request merged**